.github/workflows/continuous-integration.yml

```diff
@@ -393,7 +457,18 @@ jobs:
       - name: Create test results directory
         run: mkdir -p test-results
 
-      - name: Run Cypress tests
+      - name: Run Cypress tests on push
+        if: ${{ github.ref != 'refs/heads/master' }}
```

Comment:
Regarding your first comment, I think I'm missing a little context. But the `testing-reports` job is skipped when `needs.cypress-tests-prep.outputs.num_containers == 0`. The `if` conditional seems to work.

Reply:
Regarding your second comment, that was actually easier to implement than I thought it would be. Please check the `.github/workflows/continuous-integration.yml` and `script/github-actions/select-cypress-tests.js` files.

@darius, I think this is a cleaner solution overall (referencing my last comment to you regarding adding an `if` statement to each step in the `cypress-tests-prep` job).